src/AWS.Deploy.Orchestration/SystemCapabilityEvaluator.cs

```diff
@@ -27,105 +28,161 @@ public class SystemCapabilityEvaluator : ISystemCapabilityEvaluator
-
-                    capabilities.Add(new SystemCapability(NODEJS_DEPENDENCY_NAME, message, NODEJS_INSTALLATION_URL));
-                }
-                else if (systemCapabilities.NodeJsVersion < MinimumNodeJSVersion)
-                {
-                    message = $"Install Node.js {MinimumNodeJSVersion} or later and restart your IDE/Shell. The latest Node.js LTS version is recommended. This deployment option uses the AWS CDK, which requires Node.js version higher than your current installation ({systemCapabilities.NodeJsVersion}). ";
-
-
-                    capabilities.Add(new SystemCapability(NODEJS_DEPENDENCY_NAME, message, NODEJS_INSTALLATION_URL));
+                    var nodeInfo = await GetNodeJsVersionAsync();
+
+                    if (nodeInfo.NodeJsVersion == null)
+                    {
+                        message = $"Install Node.js {MinimumNodeJSVersion} or later and restart your IDE/Shell. The latest Node.js LTS version is recommended. This deployment option uses the AWS CDK, which requires Node.js.";
+
+                        missingCapabilitiesForRecipe.Add(new SystemCapability(NODEJS_DEPENDENCY_NAME, message, NODEJS_INSTALLATION_URL));
+                    }
+                    else if (nodeInfo.NodeJsVersion < MinimumNodeJSVersion)
+                    {
+                        message = $"Install Node.js {MinimumNodeJSVersion} or later and restart your IDE/Shell. The latest Node.js LTS version is recommended. This deployment option uses the AWS CDK, which requires Node.js version higher than your current installation ({nodeInfo.NodeJsVersion}). ";
+
+                        missingCapabilitiesForRecipe.Add(new SystemCapability(NODEJS_DEPENDENCY_NAME, message, NODEJS_INSTALLATION_URL));
+                    }
+                    else // It is valid, so update the cache interval
+                    {
+                        _nodeDependencyValidUntilUtc = DateTime.UtcNow.Add(DEPENDENCY_CACHE_INTERVAL);
+                    }
                 }
             }
 
+            // We only need to check that Docker is installed if the user is deploying a recipe that uses Docker
             if (selectedRecommendation.Recipe.DeploymentBundle == DeploymentBundleTypes.Container)
             {
-                if (!systemCapabilities.DockerInfo.DockerInstalled)
-                {
-                    message = "Install and start Docker version appropriate for your OS. This deployment option requires Docker, which was not detected.";
-                    capabilities.Add(new SystemCapability(DOCKER_DEPENDENCY_NAME, message, DOCKER_INSTALLATION_URL));
-                }
-                else if (!systemCapabilities.DockerInfo.DockerContainerType.Equals("linux", StringComparison.OrdinalIgnoreCase))
+                if (_dockerDependencyValidUntilUtc == null || DateTime.UtcNow >= _dockerDependencyValidUntilUtc)
```

Comment:
nit: Consider setting the default value of `_dockerDependencyValidUntilUtc` to [DateTime.MinValue](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.minvalue?view=net-8.0) instead on null.

Reply:
Done in https://github.com/aws/aws-dotnet-deploy/pull/837/commits/b22ed89708bf1b4e05ec4b89d774f841be431331